File: android/app/main/java/com/salman/herbalencyclopedia/model/Category.java
package com.salman.herbalencyclopedia.model;

import java.io.Serializable;

/** يمثّل تصنيفاً للأعشاب (مثل: مهدئة، هاضمة، مضادة للالتهاب...). */
public class Category implements Serializable {

    private final String id;
    private final String name;

    public Category(String id, String name) {
        this.id = id;
        this.name = name;
    }

    public String getId() { return id; }
    public String getName() { return name; }
}

File: app/src/main/java/com/salman/herbalencyclopedia/data/model/Category.kt
package com.salman.herbalencyclopedia.data.model

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.PropertyName

data class Category @JvmOverloads constructor(
    @DocumentId
    val id: String = "",

    @get:PropertyName("name")
    @set:PropertyName("name")
    var name: String = "",

    @get:PropertyName("icon")
    @set:PropertyName("icon")
    var icon: String? = null
)
